src/main/java/org/prebid/server/handler/openrtb2/AmpHandler.java

```diff
@@ -115,7 +120,10 @@ public void handle(RoutingContext context) {
                         addToEvent(result.getRight(), ampEventBuilder::bidResponse, result))
                 .map((Tuple2<BidRequest, BidResponse> result) -> toAmpResponse(result.getLeft(), result.getRight()))
                 .recover(this::updateErrorRequestsMetric)
-                .map(ampResponse -> addToEvent(ampResponse.getTargeting(), ampEventBuilder::targeting, ampResponse))
+                .compose(result ->
```

Comment:
For clarity it might be good to specify the type of `result` like this: `(Tuple3<BidRequest, BidResponse, AmpResponse> result)`